tests/test_stats.cc

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <albatross/Common>
+#include <albatross/Stats>
+#include <gtest/gtest.h>
+
+namespace albatross {
+
+TEST(test_stats, test_chi_squared) {
+
+  EXPECT_LT(fabs(chi_squared_cdf(9.260, 23) - 0.005), 1e-5);
+  EXPECT_LT(fabs(chi_squared_cdf(38.932, 21) - 0.99), 1e-5);
+  EXPECT_LT(fabs(chi_squared_cdf(96.578, 80) - 0.9), 1e-5);
+  EXPECT_LT(fabs(chi_squared_cdf(70.065, 100) - 0.01), 1e-5);
+
+  EXPECT_LT(chi_squared_cdf(0., 0.), 1e-6);
```

Comment:
This seems ill defined to me. The pdf of chi2(0) is a delta dirac at 0. This makes all positive numbers greater then the anything sampled from chi2(0), but it makes 0 equal to anything sampled from chi2(0). I would've said this should be 1.

Reply:
nice catch